sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,650 @@
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.containers.containerregistry;
+
+import com.azure.containers.containerregistry.implementation.ContainerRegistriesImpl;
+import com.azure.containers.containerregistry.implementation.ContainerRegistryImpl;
+import com.azure.containers.containerregistry.implementation.ContainerRegistryImplBuilder;
+import com.azure.containers.containerregistry.implementation.ContainerRegistryRepositoriesImpl;
+import com.azure.containers.containerregistry.implementation.models.TagAttributesBase;
+import com.azure.containers.containerregistry.models.ContentProperties;
+import com.azure.containers.containerregistry.models.DeleteRepositoryResult;
+import com.azure.containers.containerregistry.models.ListRegistryArtifactOptions;
+import com.azure.containers.containerregistry.models.ListTagsOptions;
+import com.azure.containers.containerregistry.models.RegistryArtifactProperties;
+import com.azure.containers.containerregistry.models.RepositoryProperties;
+import com.azure.containers.containerregistry.models.TagProperties;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.exception.ClientAuthenticationException;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.SerializerAdapter;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/** Initializes a new instance of the asynchronous ContainerRegistry type. */
+@ServiceClient(builder = ContainerRegistryClientBuilder.class, isAsync = true)
+public final class ContainerRepositoryAsyncClient {
+    private final ContainerRegistryRepositoriesImpl serviceClient;
+    private final ContainerRegistriesImpl registriesImplClient;
+    private final String repositoryName;
+    private final String endpoint;
+    private final String apiVersion;
+
+    private final ClientLogger logger = new ClientLogger(ContainerRepositoryAsyncClient.class);
+
+    ContainerRepositoryAsyncClient(String repositoryName, HttpPipeline httpPipeline, SerializerAdapter serializerAdapter, String endpoint, String apiVersion) {
+        if (repositoryName == null) {
+            throw logger.logExceptionAsError(new NullPointerException("'repositoryName' can't be null"));
+        }
+
+        ContainerRegistryImpl registryImpl = new ContainerRegistryImplBuilder()
+            .pipeline(httpPipeline)
+            .serializerAdapter(serializerAdapter)
```

Comment:
If this isn't passed it will implicitly default to JacksonAdapter. Shouldn't need to pass this given Container Registry doesn't need a customized SerializerAdapter